addons/mail/static/src/utils/common/hooks.js

```diff
@@ -493,3 +493,22 @@ export function useSequential() {
         return prom;
     };
 }
+
+export function useDiscussSystray() {
+    const ui = useState(useService("ui"));
+    return {
+        class: "o-mail-DiscussSystray-class",
+        get contentClass() {
+            return `d-flex flex-column flex-grow-1 bg-view ${
+                ui.isSmall ? "overflow-auto w-100 mh-100" : ""
+            }`;
+        },
+        get menuClass() {
+            return `p-0 o-mail-DiscussSystray ${
+                ui.isSmall
+                    ? "o-mail-systrayFullscreenDropdownMenu start-0 w-100 mh-100 d-flex flex-column mt-0 border-0 shadow-lg"
+                    : ""
+            }`;
+        },
+    };
+}
```

Comment:
This is component concern, it should not be inside a hook

Reply:
This is shared between `MessagingMenu` and `ActivityMenu`.
I had to change the exact same LOCs between both components in the template. This helps ensuring they reuse the same logic in their template

Reply:
I still think it should not be a hook, it's not holding any meaning full logic. It's just indirection. It should be done in the template of the component it's simpler.
It also create a dependency between those 2 components making it hard to edit them without a risk to break the other. Component should not be interdependent.

Reply:
I made a hook because it feels natural to reuse logic between 2 components with that, without the identity crisis.
Without using a hook that would mean using a lone function or object, which is unnatural with reusing code between components. Also this add another problem: this logic wants to be UI reactive with service `ui`, so it wants to make use of `useState()`. This makes a lot of sense as a hook, not a lone function.

To sum up, I disagree and it's ok to be a hook

Reply:
You miss understand me a bit, I also don't want a lone function or anything. css class logic should be done in the component template (or if complex from a component function). 
The issue is that this hook create dependencies between the components: you can't edit the border of one without editing the border of the other. I don't think it's a good thing.

Reply:
If we need separation in the future, then we will adapt the code.
For now, the parts of template of `MessagingMenu` and `ActivityMenu` were exactly the same, and they wanted to solve the exact same issue. This is definitely intentional for them to reuse the same code, so I really don't get why it's a problem for them to reuse the same code when it's exactly the intention of this change in this PR (and the current code of this parts of template confirms it)